### DISCUSSION TITLE
Analog to digital conversion  using adc  dma continuous mode

### DISCUSSION THREAD

**Discussion by valiokei** [bitcraze/discussions#1161] [Q&A]
Hi,
My idea is to execute some signal processing directly on the stm32 of the crazyflie. The signal should be acquired using an analog input pin.

After preliminary research in the examples provided within the documentation and with the ones found on the web, I still didn't have a clear idea of how an analogical signal should be converted in continuous mode using the dma to a digital signal.
I'm not interested in the gpio style analog mapping functions.

Also,  I found a lot of examples for stm32 nucleo boards, but these seem not portable directly to the crazyflie firmware.

Moreover, after the processing, the data should be usable by the ekf of the crazyflie at the maximum frequency.

**Comment by ataffanel** [comment]
This should be possible, but you will have to use the STM32 lib to achieve DMA ADC continuous mode. It is likely the lib we use is older that what is used in the new nucleo example so they would have to be ported, thought the hardware is the same so the same idea should work. One very important things to check though is to check what DMA channel is currently being used in the Crazyflie to select the one you should use, there are not documentation about what channel is used so you will have to look at the Crazyflie firmware source code.

We are using the STM32F4 STD Peripheral Library 1.8 in the Crazyflie, I could not find any documentation from ST about this lib but the header files are [pretty well documented](https://github.com/bitcraze/crazyflie-firmware/blob/master/src/lib/STM32F4xx_StdPeriph_Driver/inc/stm32f4xx_adc.h) and together with the datasheet you should be able to setup the device the way you need to.
